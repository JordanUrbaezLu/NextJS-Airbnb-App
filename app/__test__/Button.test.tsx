import { render, screen } from "@testing-library/react";
import Button from "../components/Button";

describe("Button", () => {
  test("Should render component correctly.", () => {
    const { container } = render(<Button onClick={jest.fn()} label="Airbnb" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  describe("Props", () => {
    test("Should render outline prop correctly", () => {
      render(<Button onClick={jest.fn()} label="Airbnb" outline />);

      expect(screen.getByRole("button", { name: "Airbnb" })).toHaveClass(
        "bg-white"
      );
    });

    test("Should render small prop correctly", () => {
      render(<Button onClick={jest.fn()} label="Airbnb" small />);

      expect(screen.getByRole("button", { name: "Airbnb" })).toHaveClass(
        "text-sm"
      );
    });
  });
});
