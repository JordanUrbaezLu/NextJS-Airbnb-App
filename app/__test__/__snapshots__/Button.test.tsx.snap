// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should render component correctly. 1`] = `
<button
  class="
        relative
        w-full
        rounded-lg
        transition
        hover:opacity-80
        disabled:cursor-not-allowed
        disabled:opacity-70
        bg-rose-500
        border-rose-500
        text-white
        text-md
        py-3
        font-semibold
        border-2
      "
>
  Airbnb
</button>
`;
